### PR TITLE
cleaner callback API

### PR DESCRIPTION
**Pull request by nicolaskruchten**
This PR is meant to partly address #49 and help users avoid issues like #61 or #59

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,129 +34,121 @@ $ npm install react-plotly.js plotly.js
...
+          /* data, layout, config, frames */
+        };
+    }
+
+    render() {
+        return (
+            <Plot
+                data={this.state.data}
+                layout={this.state.layout}
+                frames={this.state.frames}
```
Comment (VeraZab) [inline README.md]: so here it'd be this.state.figure.frames, this.state.figure.layout, this.state.figure.layout..right

File: README.md [hunk 2]
```diff
@@ -31,129 +34,121 @@ $ npm install react-plotly.js plotly.js
...
 
-## API
+| Prop               | Type                         | Default                                           | Description                                                                                                                                                           |
+| ------------------ | ---------------------------- | ------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `data`             | `Array`                      | `[]`                                              | list of trace objects (see https://plot.ly/javascript/reference/)                                                                                                     |
+| `layout`           | `Object`                     | `undefined`                                       | layout object (see https://plot.ly/javascript/reference/)                                                                                                             |
```
Comment (VeraZab) [inline README.md]: https://plot.ly/javascript/reference/#layout

File: README.md (hunk 1 above)
Comment (nicolaskruchten) [reply README.md]: nope, `this.setState(figure)` means that the state *is* the figure. You would be right if I'd done `this.setState({figure})` (note the `{ }`).

**Comment by VeraZab** [comment]
looks good to me! 💃

**Pull request merged**